src/utils/symlink.ts

```diff
@@ -39,29 +45,80 @@ export function createSymlinks(
   }
   const parsedOldVersion =
     (oldVersion ? parseVersion(oldVersion) : undefined) || undefined;
+
   const baseVersionName = path.basename(versionFilePath);
   const packageDir = path.dirname(versionFilePath);
 
-  // link latest, but only if the new version is "newer"
   if (
-    parsedOldVersion &&
-    !versionGreaterOrEqualThan(parsedNewVersion, parsedOldVersion)
+    !parsedOldVersion ||
+    versionGreaterOrEqualThan(parsedNewVersion, parsedOldVersion)
   ) {
-    logger.warn(
-      `Not updating the latest version file: current version is "${oldVersion}", new version is "${newVersion}"`
-    );
-  } else {
     logger.debug(
-      `Changing symlink for "latest.json" from version "${oldVersion}" to "${newVersion}"`
+      `${parsedOldVersion ? 'Updating' : 'Adding'} symlink for "latest.json" ${
+        parsedOldVersion ? `from ${oldVersion} ` : ''
+      }to "${newVersion}"`
     );
     forceSymlink(baseVersionName, path.join(packageDir, 'latest.json'));
   }
 
-  // link major
-  const majorVersionLink = `${parsedNewVersion.major}.json`;
-  forceSymlink(baseVersionName, path.join(packageDir, majorVersionLink));
+  // Read possibly existing symlinks for major and minor versions of the new version
+  const newVersionMajorSymlinkedVersion = getExistingSymlinkedVersion(
+    path.join(packageDir, `${parsedNewVersion.major}.json`)
+  );
+  const newVersionMinorSymlinkedVersion = getExistingSymlinkedVersion(
+    path.join(
+      packageDir,
+      `${parsedNewVersion.major}.${parsedNewVersion.minor}.json`
+    )
+  );
+
+  // link {major}.json if there's no link yet for that major
+  // or if the new version is newer than the currently linked one
+  if (
+    !newVersionMajorSymlinkedVersion ||
+    versionGreaterOrEqualThan(parsedNewVersion, newVersionMajorSymlinkedVersion)
+  ) {
+    const majorVersionLink = `${parsedNewVersion.major}.json`;
+    logger.debug(
+      `${
+        newVersionMajorSymlinkedVersion ? 'Updating' : 'Adding'
+      } symlink for "${majorVersionLink}" ${
+        newVersionMajorSymlinkedVersion
+          ? `from version "${semVerToString(newVersionMajorSymlinkedVersion)}" `
+          : ''
+      }to "${newVersion}"`
+    );
```

Comment:
there's way too much going on in this line -- if it's just for a debug line I'd just put the raw data in here instead of trying to make it end-user readable.  should eliminate all the logic here as well since this seems error prone / more likely to break than the actual logic

Reply:
I think `logger.debug('symlinking', {before, after})` would probably be sufficient

Reply:
updated it again